packages/next/src/server/app-render/app-render.tsx

```diff
@@ -2402,201 +2450,219 @@ async function spawnDynamicValidationInDev(
+              ServerInsertedHTMLProvider={ServerInsertedHTMLProvider}
+              ServerInsertedMetadataProvider={ServerInsertedMetadataProvider}
+              gracefullyDegrade={!!botType}
+              nonce={nonce}
+            />,
+            {
+              signal: finalClientController.signal,
+              onError: (err: unknown, errorInfo: ErrorInfo) => {
+                if (
+                  isPrerenderInterruptedError(err) ||
+                  finalClientController.signal.aborted
+                ) {
+                  const componentStack = errorInfo.componentStack
+                  if (typeof componentStack === 'string') {
+                    trackAllowedDynamicAccess(
+                      workStore.route,
+                      componentStack,
+                      dynamicValidation
+                    )
+                  }
+                  return
                 }
 
-                const componentStack = errorInfo.componentStack
-                if (typeof componentStack === 'string') {
-                  trackAllowedDynamicAccess(
-                    workStore.route,
-                    componentStack,
-                    dynamicValidation
-                  )
-                }
-                return
-              }
+                return getDigestForWellKnownError(err)
+              },
+              // We don't need bootstrap scripts in this prerender
+              // bootstrapScripts: [bootstrapScript],
+            }
+          ),
+        () => {
+          finalClientController.abort()
+        }
+      )
 
-              return getDigestForWellKnownError(err)
-            },
-          }
-        ),
-      () => {
-        finalClientController.abort()
-        serverPhasedStream.assertExhausted()
-      }
+    const { preludeIsEmpty } = await processPrelude(unprocessedPrelude)
+
+    resolveValidation(
+      <LogSafely
+        fn={throwIfDisallowedDynamic.bind(
+          null,
+          workStore,
+          preludeIsEmpty,
+          dynamicValidation,
+          serverDynamicTracking,
+          clientDynamicTracking
+        )}
+      />
     )
-    preludeIsEmpty = (await processPrelude(unprocessedPrelude)).preludeIsEmpty
-  } catch (err) {
-    rootDidError = true
-    preludeIsEmpty = true
-    if (
-      isPrerenderInterruptedError(err) ||
-      finalClientController.signal.aborted
-    ) {
-      // we don't have a root because the abort errored in the root. We can just ignore this error
-    } else {
-      // If an error is thrown in the root before prerendering is aborted, we
-      // don't want to rethrow it here, otherwise this would lead to a hanging
-      // response and unhandled rejection. We also don't want to log it, because
-      // it's most likely already logged as part of the normal render. So we
-      // just fall through here, to make sure `resolveValidation` is called.
+  } catch (thrownValue) {
```

Comment:
this is novel. in the prerenderToStream impl the throw here gets caught in the place where we produce an error page. We don't need that in this pathway and just end up considering the prelude empty